py/objobject.c

```diff
@@ -59,13 +59,48 @@ STATIC mp_obj_t object___new__(mp_obj_t cls) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(object___new___fun_obj, object___new__);
 STATIC MP_DEFINE_CONST_STATICMETHOD_OBJ(object___new___obj, MP_ROM_PTR(&object___new___fun_obj));
 
+#if MICROPY_PY_ATTRS_METHODS
+STATIC mp_obj_t object___setattr__(mp_obj_t self_in, mp_obj_t attr_in, mp_obj_t value) {
+    mp_map_lookup(mp_obj_dict_get_map(self_in), MP_OBJ_NEW_QSTR(mp_obj_str_get_qstr(attr_in)), MP_MAP_LOOKUP_ADD_IF_NOT_FOUND)->value = value;
+    return value;
+}
+MP_DEFINE_CONST_FUN_OBJ_3(object___setattr___obj, object___setattr__);
+
+STATIC mp_obj_t object___getattr__(mp_obj_t self_in, mp_obj_t attr_in) {
+    qstr attr = mp_obj_str_get_qstr(attr_in);
+    mp_map_elem_t *elem = mp_map_lookup(mp_obj_dict_get_map(self_in), MP_OBJ_NEW_QSTR(attr), MP_MAP_LOOKUP);
+    if (elem == NULL) {
+        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_AttributeError,
+            "'%s' object has no attribute '%q'",
+            mp_obj_get_type_str(self_in), attr));
+    }
+    return elem->value;
+}
+MP_DEFINE_CONST_FUN_OBJ_2(object___getattr___obj, object___getattr__);
+
+STATIC mp_obj_t object___delattr__(mp_obj_t self_in, mp_obj_t attr_in) {
+    mp_map_lookup(mp_obj_dict_get_map(self_in), MP_OBJ_NEW_QSTR(mp_obj_str_get_qstr(attr_in)), MP_MAP_LOOKUP_REMOVE_IF_FOUND);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_2(object___delattr___obj, object___delattr__);
+#endif
+
 STATIC const mp_rom_map_elem_t object_locals_dict_table[] = {
     #if MICROPY_CPYTHON_COMPAT
     { MP_ROM_QSTR(MP_QSTR___init__), MP_ROM_PTR(&object___init___obj) },
     #endif
     #if MICROPY_CPYTHON_COMPAT
     { MP_ROM_QSTR(MP_QSTR___new__), MP_ROM_PTR(&object___new___obj) },
     #endif
+    #if (MICROPY_CPYTHON_COMPAT && MICROPY_PY_ATTRS_METHODS)
```

Comment:
Ah didin't see that in the github diff. Still it's a bit confusing as now one might use `#define MICROPY_PY_ATTRS_METHODS 1` but it won't have any effect unless MICROPY_CPYTHON_COMPAT is also defined. But maybe there's no other way.